src/Storages/StorageBuffer.cpp

```diff
@@ -1089,18 +1089,13 @@ void StorageBuffer::writeBlockToDestination(const Block & block, StoragePtr tabl
 
 void StorageBuffer::backgroundFlush()
 {
+    try
     {
-        auto thread_group = ThreadGroup::createForBackgroundProcess(getContext());
```

Comment:
You wrote 
> /// For background tasks (i.e. Buffer flush) there may not be any group

But here thread group has been create for buffer flush. It did not match.